dbms/src/Storages/DeltaMerge/ColumnFile/ColumnFileTiny.cpp

```diff
@@ -327,15 +327,15 @@ PageIdU64 ColumnFileTiny::writeColumnFileData(
 
     auto data_size = write_buf.count();
     if (const auto & file_provider = dm_context.global_context.getFileProvider();
-        file_provider->isKeyspaceEncryptionEnabled())
+        unlikely(file_provider->isEncryptionEnabled(dm_context.keyspace_id)))
     {
-        const auto ep = EncryptionPath(std::to_string(dm_context.keyspace_id), "");
-        if (unlikely(!file_provider->isFileEncrypted(ep)))
+        if (const auto ep = EncryptionPath("", "", dm_context.keyspace_id);
+            unlikely(!file_provider->isFileEncrypted(ep)))
         {
             file_provider->createEncryptionInfo(ep);
         }
-        auto * data = write_buf.internalBuffer().begin();
-        file_provider->encryptPage(ep, data, data_size, page_id);
+        auto * data = write_buf.buffer().begin();
```

Comment:
Why change this to working_buffer?

Reply:
working_buffer contains the whole page data, internal_buffer may be larger, so working_buffer is more suitable.

Reply:
Is `writer_buff.buffer()` size must be equal to the size of `data_size`?

Reply:
change to read with ReadBuffer

```
auto data_size = write_buf.count();
auto buf = write_buf.tryGetReadBuffer();
if (const auto & file_provider = dm_context.global_context.getFileProvider();
    unlikely(file_provider->isEncryptionEnabled(dm_context.keyspace_id)))
{
    if (const auto ep = EncryptionPath("", "", dm_context.keyspace_id);
        unlikely(!file_provider->isFileEncrypted(ep)))
    {
        file_provider->createEncryptionInfo(ep);
    }

    char page_data[data_size];
    buf->readStrict(page_data, data_size);
    // encrypt the page data in place
    file_provider->encryptPage(dm_context.keyspace_id, page_data, data_size, page_id);
    // ReadBufferFromOwnString will copy the data, and own the data.
    buf = std::make_shared<ReadBufferFromOwnString>(std::string_view(page_data, data_size));
}
wbs.log.putPage(page_id, 0, buf, data_size, col_data_sizes);
```